src/game/obj_behaviors.c

```diff
@@ -717,59 +709,711 @@ UNUSED s32 debug_sequence_tracker(s16 debugInputSequence[]) {
-#include "behaviors/hoot.inc.c"
-#include "behaviors/beta_holdable_object.inc.c"
-#include "behaviors/bubble.inc.c"
-#include "behaviors/water_wave.inc.c"
-#include "behaviors/explosion.inc.c"
-#include "behaviors/respawner.inc.c"
-#include "behaviors/bully.inc.c"
-#include "behaviors/water_ring.inc.c"
-#include "behaviors/bowser_bomb.inc.c"
-#include "behaviors/celebration_star.inc.c"
-#include "behaviors/drawbridge.inc.c"
-#include "behaviors/bomp.inc.c"
-#include "behaviors/sliding_platform.inc.c"
-#include "behaviors/moneybag.inc.c"
-#include "behaviors/bowling_ball.inc.c"
-#include "behaviors/cruiser.inc.c"
-#include "behaviors/spindel.inc.c"
-#include "behaviors/pyramid_wall.inc.c"
-#include "behaviors/pyramid_elevator.inc.c"
-#include "behaviors/pyramid_top.inc.c"
-#include "behaviors/sound_waterfall.inc.c"
-#include "behaviors/sound_volcano.inc.c"
-#include "behaviors/castle_flag.inc.c"
-#include "behaviors/sound_birds.inc.c"
-#include "behaviors/sound_ambient.inc.c"
-#include "behaviors/sound_sand.inc.c"
-#include "behaviors/castle_cannon_grate.inc.c"
-#include "behaviors/snowman.inc.c"
-#include "behaviors/boulder.inc.c"
-#include "behaviors/cap.inc.c"
-#include "behaviors/koopa_shell.inc.c"
-#include "behaviors/spawn_star.inc.c"
-#include "behaviors/red_coin.inc.c"
-#include "behaviors/hidden_star.inc.c"
-#include "behaviors/rolling_log.inc.c"
-#include "behaviors/mushroom_1up.inc.c"
-#include "behaviors/controllable_platform.inc.c"
-#include "behaviors/breakable_box_small.inc.c"
-#include "behaviors/snow_mound.inc.c"
-#include "behaviors/floating_platform.inc.c"
-#include "behaviors/arrow_lift.inc.c"
-#include "behaviors/orange_number.inc.c"
-#include "behaviors/manta_ray.inc.c"
-#include "behaviors/falling_pillar.inc.c"
-#include "behaviors/floating_box.inc.c"
-#include "behaviors/decorative_pendulum.inc.c"
-#include "behaviors/treasure_chest.inc.c"
-#include "behaviors/mips.inc.c"
-#include "behaviors/yoshi.inc.c"
+s32 obj_is_rendering_enabled(void) {
+    if (o->header.gfx.node.flags & GRAPH_RENDER_ACTIVE) {
+        return TRUE;
+    } else {
+        return FALSE;
+    }
+}
+
+s16 obj_get_pitch_from_vel(void) {
+    return -atan2s(o->oForwardVel, o->oVelY);
+}
+
+/**
+ * Show dialog proposing a race.
+ * If the player accepts the race, then leave time stop enabled and Mario in the
+ * text action so that the racing object can wait before starting the race.
+ * If the player declines the race, then disable time stop and allow Mario to
+ * move again.
+ */
+s32 obj_update_race_proposition_dialog(s16 dialogID) {
+    s32 dialogResponse = cur_obj_update_dialog_with_cutscene(MARIO_DIALOG_LOOK_UP,
+        (DIALOG_FLAG_TURN_TO_MARIO | DIALOG_FLAG_TIME_STOP_ENABLED), CUTSCENE_RACE_DIALOG, dialogID);
+
+    if (dialogResponse == DIALOG_RESPONSE_NO) {
+        set_mario_npc_dialog(MARIO_DIALOG_STOP);
+        disable_time_stop_including_mario();
+    }
+
+    return dialogResponse;
+}
+
+void obj_set_dist_from_home(f32 distFromHome) {
```

Comment:
All small functions like this should be `static inline` in header files.

Reply:
I did enough function moving here to get rid of `obj_behaviors_2`, but anything further is so out of scope that i have another in progress branch trying to consolidate and split everything in a way that makes sense

Reply:
I agree, let's keep this PR majorly to splitting behaviors to C files first to ease reviewing process. Can consider minor perf stuff later.